lighthouse-core/gather/driver.js

```diff
@@ -506,10 +543,17 @@ class Driver {
     const maxWaitMs = (options.flags && options.flags.maxWaitForLoad) ||
         Driver.MAX_WAIT_FOR_FULLY_LOADED;
 
-    return this.sendCommand('Page.enable')
+    const redirectTracking = {url};
+
+    return this._beginNetworkMonitoring(redirectTracking)
+      .then(_ => this.sendCommand('Page.enable'))
       .then(_ => this.sendCommand('Emulation.setScriptExecutionDisabled', {value: disableJS}))
       .then(_ => this.sendCommand('Page.navigate', {url}))
-      .then(_ => waitForLoad && this._waitForFullyLoaded(pauseAfterLoadMs, maxWaitMs));
+      .then(_ => waitForLoad && this._waitForFullyLoaded(pauseAfterLoadMs, maxWaitMs))
+      .then(_ => {
+        this._endNetworkMonitoring();
+        return redirectTracking.url;
```

Comment:
since we're cleaning this up to not touch `options` it feels weird to still do this object mutation, what if it were restructured slightly so that we called `_monitorUrlChanges()` and could call `getLatestUrl()` at any point